client/src/featureform/enums.py

```diff
@@ -103,6 +103,9 @@ class SourceType(str, Enum):
 class FilePrefix(Enum):
     S3 = "s3://"
     S3A = "s3a://"
+    HDFS = "hdfs://"
+    GCS = "gs://"
+    AZURE = "abfss://"
```

Comment:
you could these

```
class FilePrefix(Enum):
    S3 = ("s3://", ["s3://", "s3a://"])
    S3A = ("s3a://", ["s3a://"])
    HDFS = ("hdfs://", ["hdfs://"])
    GCS = ("gs://", ["gs://"])
    AZURE = ("abfss://", ["abfss://"])

    def __init__(self, prefix: str, valid_prefixes: list[str]):
        self.prefix = prefix
        self.valid_prefixes = valid_prefixes

    def validate_file_scheme(self, file_path: str) -> (bool, str):
        has_prefix = any(file_path.startswith(pref) for pref in self.valid_prefixes)
        required_prefix = " or ".join(self.valid_prefixes)
        return has_prefix, required_prefix
```

or this for backwards compatability

```
class FilePrefix(Enum):
    S3 = "s3://"
    S3A = "s3a://"
    HDFS = "hdfs://"
    GCS = "gs://"
    AZURE = "abfss://"

    @property
    def valid_prefixes(self):
        return {
            FilePrefix.S3: ["s3://", "s3a://"],
            FilePrefix.S3A: ["s3a://"],
            FilePrefix.HDFS: ["hdfs://"],
            FilePrefix.GCS: ["gs://"],
            FilePrefix.AZURE: ["abfss://"],
        }.get(self, [self.value])

    def validate_file_scheme(self, file_path: str) -> (bool, str):
        has_prefix = any(file_path.startswith(pref) for pref in self.valid_prefixes)
        required_prefix = " or ".join(self.valid_prefixes)
        return has_prefix, required_prefix
```

The idea being you want it to be contained in here otherwise someone will change the code here and then forget the side effect in a separate function